test/Autofac.Specification.Test/Registration/LambdaGenericOverloadRegistrationTests.cs

```diff
@@ -0,0 +1,588 @@
+﻿// Copyright (c) Autofac Project. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Reflection;
+using Autofac.Core;
+
+namespace Autofac.Specification.Test.Registration;
+
+#nullable enable
+
+public class LambdaGenericOverloadRegistrationTests
+{
+    private class MyDep1
+    {
+    }
+
+    private class MyDep2
+    {
+    }
+
+    private class MyDep3
+    {
+    }
+
+    private class MyDep4
+    {
+    }
+
+    private class MyDep5
+    {
+    }
+
+    private class MyDep6
+    {
+    }
+
+    private class MyDep7
+    {
+    }
+
+    private class MyDep8
+    {
+    }
+
+    private class MyDep9
+    {
+    }
+
+    private class MyDep10
+    {
+    }
+
+    private class MyComponent
+    {
+        public MyComponent(MyDep1 d1)
+        {
+        }
+
+        public MyComponent(MyDep1 d1, MyDep2 d2)
+        {
+        }
+
+        public MyComponent(MyDep1 d1, MyDep2 d2, MyDep3 d3)
+        {
+        }
+
+        public MyComponent(MyDep1 d1, MyDep2 d2, MyDep3 d3, MyDep4 d4)
+        {
+        }
+
+        public MyComponent(MyDep1 d1, MyDep2 d2, MyDep3 d3, MyDep4 d4, MyDep5 d5)
+        {
+        }
+
+        public MyComponent(MyDep1 d1, MyDep2 d2, MyDep3 d3, MyDep4 d4, MyDep5 d5, MyDep6 d6)
+        {
+        }
+
+        public MyComponent(MyDep1 d1, MyDep2 d2, MyDep3 d3, MyDep4 d4, MyDep5 d5, MyDep6 d6, MyDep7 d7)
+        {
+        }
+
+        public MyComponent(MyDep1 d1, MyDep2 d2, MyDep3 d3, MyDep4 d4, MyDep5 d5, MyDep6 d6, MyDep7 d7, MyDep8 d8)
+        {
+        }
+
+        public MyComponent(MyDep1 d1, MyDep2 d2, MyDep3 d3, MyDep4 d4, MyDep5 d5, MyDep6 d6, MyDep7 d7, MyDep8 d8, MyDep9 d9)
+        {
+        }
+
+        public MyComponent(MyDep1 d1, MyDep2 d2, MyDep3 d3, MyDep4 d4, MyDep5 d5, MyDep6 d6, MyDep7 d7, MyDep8 d8, MyDep9 d9, MyDep10 d10)
+        {
+        }
+    }
+
+    private class MyComponentWithParams
+    {
+        public MyComponentWithParams(MyDep1 dep1, string arg1, string arg2)
+        {
+            Arg1 = arg1;
+            Arg2 = arg2;
+        }
+
+        public string Arg1 { get; }
+
+        public string Arg2 { get; }
+    }
+
+    private static readonly Type[] _allDeps = new[]
```

Comment:
Will update the name; I'd suggest that if an editorconfig is down as a suggestion, but it would fail at code review, it should be a warning rather than a suggestion.

I read suggestions as "personal preference on not doing X, or if you know there's a good reason not to do X", rather than "don't do X".

I've updated the editorconfig to set `constant_fields_should_be_pascal_case, `static_fields_should_be_camel_case`, `static_readonly_fields_should_be_pascal_case` and `camel_case_for_private_internal_fields` to `warning`, and no additional warnings have been raised, so seems like we should do that.